ignite/handlers/checkpoint.py

```diff
@@ -295,7 +298,17 @@ def __call__(self, engine: Engine) -> None:
                 for k in checkpoint:
                     name = k
                 checkpoint = checkpoint[name]
-            filename = "{}{}_{}{}".format(self._fname_prefix, name, suffix, self._ext)
+
+            filename_pattern_dict["name"] = name
+            filename_pattern_dict["filename_prefix"] = self.filename_prefix
+            filename_pattern_dict["ext"] = self._ext
+            if Checkpoint.filename_pattern != Checkpoint.default_filename_pattern:
```

Comment:
IMO, code L305-L311 looks a bit ugly. We can in all cases factorize `filename = Checkpoint.filename_pattern.format(**filename_pattern_dict)`. And this check is needed to apply removal of `_` if some of pattern keys were not used. 
Let's think if we can simplify this part somehow and also remove IMO redundant class attribute `default_filename_pattern` ...

Reply:
You are right @vfdev-5 its a bit ugly. Let me try to change it a bit to remove the redundant class attribute

Reply:
```suggestion
            filename_pattern_dict["name"] = name

            filename = Checkpoint.filename_pattern.format(**filename_pattern_dict)

            if not self.filename_prefix:
                filename = filename.replace("_{}".format(name), name)

            if self.global_step_transform is None:
                filename = filename.replace("__", "_")

            if self._score_name is None:
                filename = filename.replace("_=", "_")

            if self._score_name is None and not filename_pattern_dict["score"]:
                filename = filename.replace("_.", ".")
```
Will this work?